core/corehttp/gateway_handler.go

```diff
@@ -125,6 +125,10 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	i.addUserHeaders(w) // ok, _now_ write user's headers.
 	w.Header().Set("X-IPFS-Path", urlPath)
 
+	//robots-header to prevent search-engines from indexing the gateway because of the duplicate-content problematic when makeing the gateway public
+	w.Header().Set("X-Robots-Tag", "noindex")
```

Comment:
>  I can tell you that google assigns every [a-z0-9]{16}.onion domain to exactly one of this tor2web-proxies. The same subdomain at any other of the proxies is indexed but is not realy ranking for anything.
> If you have more than one ipfs-gateway crawlable than google would also take the one with the most trust and unrank every other one because there a no subdomains do differentiate.

great! this is what we want.

> The proposed sitemap should in no way include everything from ipfs. It should be fillable by the gateway-owner to show only what belongs to him.

I think this is a separate concern then -- we're discussing different pieces. I'm describing behavior for a full HTTP-to-IPFS gateway, a node existing only as a bridge between the networks. it sounds you want a more limited one -- specific to a user. And in such a case, authors may also want to explicitly only allow certain content to be viewed + served-- much like how the gateway for the `localhost:5001` server works (currently only serving webui). (and down the road maybe include auth in the process)